src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,31 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int MAX_COUNT_OF_LETTERS = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        String line;
+        StringBuilder concatenateInput = new StringBuilder();
+        Set<Character> set = new TreeSet<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while ((line = reader.readLine()) != null) {
+                for (int i = 0; i < line.length(); i++) {
```

Comment:
let's try removing the `for` loop
we have a way to read per one character in `while` loop